docs/content/sips/016-component-imports.md

```diff
@@ -0,0 +1,53 @@
+title = "SIP 016 - Spin Component Imports" 
+template = "main" 
+date = "2023-09-22T16:27:43Z"
+---
+
+Summary: A proposal to introduce component imports to the Spin 2.0 manifest.
+
+Owner: <brian.hardock@fermyon.com>
+
+Created: Sep 19, 2023
+
+## Background / Goals
+The Spin 2.0 framework will support executing preview2 WebAssembly components. These components will include `import` and `export` statements. This manifest proposal is to enable Spin developers the ability to expressly configure how to satisfy the imports of their components. Furthermore, developers should be enabled to use the exports of a component to satisfy the imports of another.
+
+## Proposal
+
+### Changes to Spin Manifest 
+To support component imports, a new section will be added to the Spin 2.0 manifest design, `[component.<component-id>.import.<import-id>]`. The following sections describe the various options for specifying how to satisfy imports.
```

Comment:
For simple use cases this could be achieved through static component composition (i.e., taking two components and making them 1 component). However ,there are pros and cons to this approach as opposed to satisfying imports by instantiating the other component and using the instantiated instance in the `Host` implementation (what I call "dynamic composition"). 

**Pros of static composition (compared to "dynamic composition")**
* **It's more performant**: there's a penalty paid for exciting the guest and entering the host. Static composition avoids this cost since both components are now the same 1 guest component instead of two separate guest instances.
* **It can be shared**: static composition creates a single new component that can be shared through a component registry and even run in other environments. Dynamic composition requires a Spin runtime to hook everything up at runtime. 

**Cons of static composition**
* **it's less flexible**: dynamic composition allows the user to compose components in different ways. As an off the cuff example: imagine a "request sampling" component that we wouldn't to intercept every 10th request to do some sort of sampled metrics but otherwise not touch the other 9 requests. That's not possible with static composition but would be with a host aware dyanmic composition. 

I think in general we probably don't want the user to know or care about which approach we take. Exposing composed components that the user can use outside of `spin up` should be a separate feature IMO.